lib/src/edhoc.rs

```diff
@@ -287,19 +289,86 @@ pub fn r_process_message_3(
             let decoded_p3_res = decode_plaintext_3(&plaintext_3);
 
             if decoded_p3_res.is_ok() {
-                let (kid, mac_3, ead_3) = decoded_p3_res.unwrap();
+                let (id_cred_i, mac_3, ead_3) = decoded_p3_res.unwrap();
```

Comment:
Just to leave this as a note, for now this will only work for the following cases:
1. R receives KID and has the associated CRED_I locally stored
2. R receives CRED_I and uses it

According to `lake-authz`, R (V) could also support receiving KID and calling W to resolve CRED_I. We could leave a TO-DO for that in the code.

Reply:
done